tests/streamlit_ui_tests/streamlit.spec.ts

```diff
@@ -300,6 +380,220 @@ test.describe('Home Tab Buttons', () => {
+    await mito.getByRole('button', { name: 'None' }).click();
+    await awaitResponse(page);
+
+    await expect(mito.locator('.rangeslider-slidebox')).not.toBeVisible();
+
+    await mito.getByRole('button', { name: '▾ Add Chart Element' }).click();
+    await mito.getByRole('button', { name: 'Show Range Slider' }).hover();
+    await mito.getByRole('button', { name: 'Horizontal' }).click();
+    await awaitResponse(page);
+
+    await expect(mito.locator('.rangeslider-slidebox')).toBeVisible();
+  });
+
+  test('Hide and show gridlines', async ({ page }) => {
+    const mito = await getMitoFrameWithTypeCSV(page);
+
+    // Check that the gridlines are visible (by default)
+    await clickButtonAndAwaitResponse(page, mito, { name: 'Graph' })
+    await expect(mito.locator('.xgrid.crisp')).toHaveCount(5);
+    await expect(mito.locator('.ygrid.crisp')).toHaveCount(4);
+
+    // Hide the y gridlines
+    await mito.getByRole('button', { name: '▾ Add Chart Element' }).click();
+    await mito.getByRole('button', { name: 'Grid Lines' }).hover();
+    await mito.getByRole('button', { name: 'Horizontal' }).click();
+    await awaitResponse(page);
+
+    // check that the y gridlines are hidden and the x gridlines are not affected
+    await expect(mito.locator('.ygrid.crisp')).toHaveCount(0);
+    await expect(mito.locator('.xgrid.crisp')).toHaveCount(5);
+
+    // Hide the x gridlines
+    await mito.getByRole('button', { name: '▾ Add Chart Element' }).click();
+    await mito.getByRole('button', { name: 'Grid Lines' }).hover();
+    await mito.getByRole('button', { name: 'Vertical' }).click();
+    await awaitResponse(page);
+
+    // check that the x gridlines are hidden and the y gridlines are not affected
+    await expect(mito.locator('.ygrid.crisp')).toHaveCount(0);
+    await expect(mito.locator('.xgrid.crisp')).toHaveCount(0);
+
+    // Show the y gridlines
+    await mito.getByRole('button', { name: '▾ Add Chart Element' }).click();
+    await mito.getByRole('button', { name: 'Grid Lines' }).hover();
+    await mito.getByRole('button', { name: 'Horizontal' }).click();
+    await awaitResponse(page);
+
+    // Check that the y gridlines are visible and the x gridlines are not affected
+    await expect(mito.locator('.ygrid.crisp')).toHaveCount(4);
+    await expect(mito.locator('.xgrid.crisp')).toHaveCount(0);
+
+    // Show the x gridlines
+    await mito.getByRole('button', { name: '▾ Add Chart Element' }).click();
+    await mito.getByRole('button', { name: 'Grid Lines' }).hover();
+    await mito.getByRole('button', { name: 'Vertical' }).click();
+    await awaitResponse(page);
+
+    // Check that the x gridlines are visible and the y gridlines are not affected
+    await expect(mito.locator('.ygrid.crisp')).toHaveCount(4);
+    await expect(mito.locator('.xgrid.crisp')).toHaveCount(5);
+  });
+
+  test('Escape key closes graph title input', async ({ page }) => {
```

Comment:
Some other test ideas:

- [ ] Setting the format of the graph
- [ ] Is it possible to click the copy export button and then check that the code is copied to the clipboard?
- [ ] Create a graph, close the select data taskpane, reopen it, add a second trace and make sure graph renders. Maybe we can check that the legend is created
- [ ] Make a histogram and then change the aggregation function and check that the y axis label updates 
- [ ] Make a graph, switch to the data tab, switch back to the graph tab and the graph should re-render 
- [ ] Make a graph with column A on the X axis, switch back to the data tab, apply a filter to remove some of the strings from column A, reopen the graph and the filtered out data should no longer be in the graph
- [ ] Open the select sheet taskpane, open full screen mode, close full screen mode, make sure you can still click on the X button to close the taskpane